diwan/src/main.rs

```diff
@@ -0,0 +1,46 @@
+mod commands;
+mod editor;
+mod utils;
+
+use editor::Editor;
+use utils::FileManager;
+use log::{info, error};
+use env_logger;
+use tokio;
+
+#[tokio::main]
+async fn main() -> std::io::Result<()> {
```

Comment:
```suggestion
use anyhow::{Context, Result};

#[tokio::main]
async fn main() -> Result<Self, anyhow::Error> {
```

It is good practice to start using `anyhow` for error handling and use `.context("content error")` instead of `?` for more clear error and debugging.